src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/service/ReferralService.kt

```diff
@@ -141,14 +141,18 @@ class ReferralService(
     overrideCreatedAt: OffsetDateTime? = null,
     endOfServiceReport: EndOfServiceReport? = null,
   ): Referral {
+    val intervention = interventionRepository.getOne(interventionId)
```

Comment:
what's the reason to choose `getOne` over `findOne` in this instance?

Reply:
and what happens if the intervention id is invalid? perhaps we should throw ServerWebInputException?

``` 
val intervention = interventionRepository.findByIdOrNull(interventionId)
      ?: throw ServerWebInputException("invalid intervention id")
```

Reply:
getOne throws EntityNotFoundException which the ReferralController catches and rethrows it as the exception you described above.

I think that ServerWebInputException is outside the scope of this class?

Reply:
👍